packages/sitecore-jss/src/i18n/graphql-dictionary-service.test.ts

```diff
@@ -270,4 +271,87 @@ describe('GraphQLDictionaryService', () => {
     expect(calledWithArgs.retries).to.equal(mockServiceConfig.retries);
     expect(calledWithArgs.retryStrategy).to.deep.equal(mockServiceConfig.retryStrategy);
   });
+
+  describe('with site query', () => {
+    it('should fetch dictionary phrases using clientFactory', async () => {
+      nock(endpoint, { reqheaders: { sc_apikey: apiKey } })
+        .post('/')
+        .reply(200, dictionarySiteQueryResponse.singlepage);
+
+      const service = new GraphQLDictionaryService({
+        siteName,
+        cacheEnabled: false,
+        clientFactory,
+        useSiteQuery: true,
+      });
+      const result = await service.fetchDictionaryData('en');
+      expect(result.foo).to.equal('foo');
+      expect(result.bar).to.equal('bar');
+    });
+
+    it('should use default pageSize, if pageSize not provided', async () => {
+      nock(endpoint)
+        .persist()
+        .post(
+          '/',
+          (body) =>
+            body.query.indexOf('$pageSize: Int = 10') > 0 && body.variables.pageSize === undefined
```

Comment:
We need to bump our default page size (a primary goal with this change is to reduce calls to edge). I believe 10 was the default for `search` but for `dictionary` the default is 500 and maximum is 1000. I'd suggest 500.